src/Modules/CalcDefence.lua

```diff
@@ -2034,24 +2083,73 @@ function calcs.defence(env, actor)
 		end
 	end
 	
+	-- recoup
+	if output["anyRecoup"] > 0 then
```

Comment:
This should be re writeable with a for loop which should generalise some of this and be a bit neater.